mlflow/sklearn.py

```diff
@@ -48,25 +61,36 @@ def save_model(sk_model, path, conda_env=None, mlflow_model=Model()):
     >>> sk_path_dir = ...
     >>> mlflow.sklearn.save_model(sk_model, sk_path_dir)
     """
+    if serialization_format not in SUPPORTED_SERIALIZATION_FORMATS:
+        raise MlflowException(
+                message=(
+                    "Unrecognized serialization format: {serialization_format}. Please specify one"
+                    " of the following supported formats: {supported_formats}.".format(
+                        serialization_format=serialization_format,
+                        supported_formats=SUPPORTED_SERIALIZATION_FORMATS)),
+                error_code=INVALID_PARAMETER_VALUE)
+
     if os.path.exists(path):
         raise Exception("Path '{}' already exists".format(path))
     os.makedirs(path)
-    model_file = os.path.join(path, "model.pkl")
-    with open(model_file, "wb") as out:
-        pickle.dump(sk_model, out)
+    model_data_subpath = "model.pkl"
```

Comment:
`model_data_subpath` variable name sounds more confusing.
`model_file` or `model_file_name` was more explicit what this was doing...

Reply:
I disagree. `model_data_subpath` is meant to indicate that the specified path is a relative path to model data. `model_file` seems more generic: for example, we refer to model configurations (`mlflow.models.Model`) as "models", yet that's not what "model.pkl" is referring to. Additionally, defining this path as a variable seems preferable to using multiple instances of the inline "model.pkl" string.

Let me know if there's a variable name that you would prefer.